url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/TestForHttpHeader.java

```diff
@@ -0,0 +1,37 @@
+package net.hetimatan.util.http;
+
+import java.io.IOException;
+import net.hetimatan.io.file.MarkableFileReader;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.util.http.HttpRequestHeader;
+import junit.framework.TestCase;
+
+public class TestForHttpHeader extends TestCase {
+
+	public void testHello() {
+
+	}
+
+	public void testEncode001() throws IOException {
+		HttpRequestHeader header = new HttpRequestHeader("key","value");
+		CashKyoroFile output = new CashKyoroFile(512);
+		header.encode(output.getLastOutput());
+
+		output.seek(0);
+		byte[] buffer = new byte[(int)output.length()];
+		int len = output.read(buffer);
+		String tag = new String(buffer, 0, len);
+		assertEquals("key: value\r\n", tag);
+	}
+
+	public void testDecode001() throws IOException {
+		CashKyoroFile base = new CashKyoroFile(512);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *base*. Currently, there are execution paths that do not contain closure statements, for example, when CashKyoroFile.addChunk throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).